src/batch/src/executor/join/hash_join.rs

```diff
@@ -531,8 +531,12 @@ impl<K: HashKey> HashJoinExecutor<K> {
                         probe_row,
                         build_data_types.len(),
                     ) {
-                        non_equi_state.first_output_row_id.clear();
-                        yield spilled
+                        yield Self::process_left_outer_join_non_equi_condition(
```

Comment:
Something that confuses me is that this is from the `left outer` part, i.e. the part with no matches.
We only hit this branch when probe side does not find matches with the build side.

As such wouldn't the `where is not null` clause filter all these out, even if there were duplicates?

Reply:
This `process_left_outer_join_non_equi_condition` applies on a chunk. There are other row not from the unmatched result.

Reply:
Let me see if I got it. So we have a `chunk_builder` which is shared by both `matched` and `unmatched` rows.
So when `chunk_builder` appends row from the `unmatched` side, `matched` rows can be part of what is spilled.

Because `matched` rows can be duplicated, we need to apply `process_left_outer_join...` to them since that contains deduplication logic.

Reply:
Yes exactly